protocols/grpc/agent.proto

```diff
@@ -309,6 +310,18 @@ message OnlineCPUMemRequest {
 	// CpuOnly specifies whether only online CPU or not.
 	bool cpu_only = 3;
 }
+message OfflineCPUMemRequest {
+	// Wait specifies if the caller waits for the agent to online all resources.
```

Comment:
/to online/to offline/

Reply:
> @justin-he I have similar questions as @sboeuf.
> I would like to understand the approach you are taking towards handling CPUs on arm, mainly interaction between runtime and the agent for CPU requests and how this differs from x86_64.
> Can you add more details about the design?

Sure.
The target is to support cpu limitation for “docker run” cmd in Kata-containers project
e.g.
$ docker run --runtime=kata-runtime -ti --cpus 2 debian bash
$ docker update --cpus 4 foo
$ docker update --cpus 2 foo
Difficulties:
The best way to enable cpu hotplug for kata-containers is supporting acpi driven cpu hotplug in qemu guest. But there are some obstacles which can’t be overcome.
- The apci driven cpu hotplug is not supported in qemu guest till now, and even no clear plan for the future as far as I know.
- limitation at kvm side for gic to handle hot-adding cpu after booting. 
Mentioned by Mark Rutland:
>We'd need something along those lines. Each CPU has a notional point to
>point link to the interrupt controller (to the redistributor, to be
>precise), and this entity must pre-exist.

Thus we need to find some workaround.

My proposals:
qemu cpu hotplug(adding) process, i.e. create a cpus pool for the qemu guest
 ```
+---------------+               +----------------+         +-----------------+
 |   kata qmp    |               |     qemu       |         |     guest       |
 +-------+-------+               +----------------+         +--------+--------+
         |                      start guest with -smp 4              |
         |                       -append “maxcpus=1”                 |
         |                                |                          |
         |                                +------------------------->|
         |                                |                    start cpu with
         |                                |                   1 online,3 offline
         |                                |                          |
         |  query_hotpluggable_cpus       |                          |
         +-------------------------------->                          |
         |                                |                          |
         | return cpu node info           |                          |
         +<-------------------------------+                          |
         |                                |                          |
         |      device_add cpu            |                          |
         +------------------------------->+                          |
         |                                |                          |
         |                 online/offline cpus by gRPC               |
         +---------------------------------------------------------->+
         |                                |                          |
         +                                +                          +
```
1. When changed qemu run -smp 4 -append “maxcpus=1”, It can create 4 cpus in 
   device-tree, and set 3 as pluggable. Actually, now there are 4 cpu nodes in 
   the device-tree. This work around the vcpu re-init obstacle at kvm side.
2. Then the guest kernel will be boot with 4 cpus, but let 3 be offline
3. use qmp cmd "query_hotpluggable_cpus" to get the pluggable cpus info in the
   guest.
4. When qmp cmd "device_add" is triggered, guest will be notified to set one 
   pluggable cpus as online.
5. Use gRPC req “onlineCPU” to set the wanted cpus to online at kata agent side.
6. Use gRPC req “offlineCPU” to set the wanted cpus to offline at kata agent side.